scormxblock/scorm_file_uploader.py

```diff
@@ -135,32 +146,43 @@ def _cleanup_storage_dir(self, storage):
                 except AttributeError:
                     raise
 
-    def _extract_zipped_file(self):
-        zip_file = zipfile.ZipFile(self.temp_file_path, 'r')
-        tempdir = tempfile.mkdtemp()
-        zip_file.extractall(tempdir)
+    def _files_to_store(self, tempdir):
+        files_to_store = []
+        total_files_size = 0
 
-        return tempdir
+        for (dirpath, dirnames, files) in os.walk(tempdir):
+            for f in files:
+                file_path = os.path.join(os.path.abspath(dirpath), f)
+                size = os.path.getsize(file_path)
+                total_files_size += size
+                files_to_store.append({'path': file_path, 'size': size})
 
-    def _write_to_file(self, mode):
-        with open(self.temp_file_path, mode) as temp_file:
-            for chunk in self.scorm_file.chunks():
-                temp_file.write(chunk)
+        return files_to_store, total_files_size
 
-    def _get_content_range(self):
+    def _set_upload_progress(self, uploaded, total):
+        percent = int(uploaded / float(total) * 100)
+        block_id = self.xblock.location.block_id
+
+        cache_key = ScormPackageUploader.get_progress_cache_key(block_id)
+        cache.set(cache_key, percent, 1 * 60 * 60)
+
+    def _post_upload_cleanup(self, tempdir):
         try:
-            matches = CONTENT_RE.search(self.request.headers['Content-Range'])
-            content_range = matches.groupdict()
-        except KeyError:  # Single chunk
-            # no Content-Range header, so make one that will work
-            content_range = {'start': 0, 'stop': 1, 'end': 2}
+            shutil.rmtree(tempdir)
+            os.remove(self.temp_file_path)
+        except Exception:
+            pass
 
-        return content_range
+    @staticmethod
+    def get_progress_cache_key(block_id):
```

Comment:
Should this method be exposed outside of this class? Since this cache key is very specific to scorm upload class and being used only inside of this class; it should be private method instead IMO.